### PR TITLE
updates create travis token endpoint to read github token from database and exchange token isntead of posting token

### PR DESCRIPTION
**Pull request by jackcmeyer**
* adds the ability to exchange github token for a travis token instead of posting a travis token

### CODE REVIEW

File: src/controllers/travisauth.controller.js [hunk 1]
```diff
@@ -34,9 +34,9 @@ export async function addTravisTokenForUser(req, res, next) {
   }
 
   const userId = req.user.id;
-  const token = req.body.travisToken;
+  const githubToken = req.body.githubToken; // travis exchanges github token for travis token
```
Comment (clarkerican) [inline src/controllers/travisauth.controller.js]: This githubToken does not exist/is not used

File: src/services/travisauth.service.js [hunk 2]
```diff
@@ -55,28 +60,66 @@ export default class TravisAuthenticationService {
...
+   */
+  static async exchangeGithubTokenForTravisToken(githubToken) {
+    const url = 'https://api.travis-ci.org/auth/github';
+
+    try {
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {
+          'User-Agent': 'Travis/1.0',
+          Accept: 'application/vnd.travis-ci.2+json',
```
Comment (clarkerican) [inline src/services/travisauth.service.js]: Does Accept need to be in single quotes as well?

File: src/services/travisauth.service.js [hunk 3]
```diff
@@ -55,28 +60,66 @@ export default class TravisAuthenticationService {
...
     }
 
     const tokenToBeCreated = {
-      token: encryptUtil.encrypt(token.toString())
+      token: encryptUtil.encrypt(travisToken.toString())
     };
 
     const tokenCreated = await this.travisTokenRepository.create(tokenToBeCreated);
     await user.addTravisToken(tokenCreated);
+
```
Comment (clarkerican) [inline src/services/travisauth.service.js]: One space here, not two.

File: src/controllers/travisauth.controller.js [hunk 4]
```diff
@@ -34,9 +34,9 @@ export async function addTravisTokenForUser(req, res, next) {
...
   try {
-    await travisAuthService.addTravisTokenForUser(userId, token);
+    await travisAuthService.addTravisTokenForUser(userId, githubToken);
```
Comment (clarkerican) [inline src/controllers/travisauth.controller.js]: Remove githubToken

File: src/controllers/travisauth.controller.js (hunk 1 above)
Comment (jackcmeyer) [reply src/controllers/travisauth.controller.js]: this isn't needed, I'll remove it

File: src/services/travisauth.service.js (hunk 2 above)
Comment (jackcmeyer) [reply src/services/travisauth.service.js]: Nope. If it is a one word without any funny characters, it don't need quotes. linter will catch this

**Pull request merged**